ambry-api/src/main/java/com.github.ambry/account/ContainerBuilder.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static com.github.ambry.account.Container.*;
+
+
+/**
+ * A builder class for {@link Container}. Since {@link Container} is immutable, modifying a {@link Container} needs to
+ * build a new {@link Container} object with updated fields through this builder. A {@link Container} can be built
+ * in two ways: 1) from an existing {@link Container} object; and 2) by supplying required fields of a {@link Container}.
+ */
+public class ContainerBuilder {
+  private Short id;
+  private String name;
+  private ContainerStatus status;
+  private String description;
+  private Boolean isPrivate;
+  private Short parentAccountId;
+  private short version;
+
+  /**
+   * Constructor. This will allow building a new {@link Container} from an existing {@link Container}. The builder will
+   * include all the information of the existing {@link Container}.
+   * @param origin The {@link Container} to build from.
+   */
+  public ContainerBuilder(Container origin) {
+    if (origin == null) {
+      throw new IllegalArgumentException("origin cannot be null.");
+    }
+    this.id = origin.getId();
```

Comment:
You have brought a good point. The logic in `AccountBuilder` and `ContainerBuilder` has been changed to always construct into an object in the highest version, and always serialize into json in the highest version.